tutorials/install-pgbouncer/index.mdx

```diff
@@ -0,0 +1,252 @@
+- `RESUME [DB]`: Resumes the `KILL`, `PAUSE`, or `SUSPEND` the specific database.
+- `RECONNECT [DB]`: Closes and reconnects all server connections. If you are planning to do a switchover, consider using `PAUSE` command. If you have done a failover, consider using the `KILL` command.
+- `DISABLE [DB]`: Disallows new connections to the specific database.
+- `ENABLE [DB]`: Allows new connections to the specific database after a `DISABLE` command.
+- `KILL [DB]`: Drops all client and server connections and pauses the specific database.
+- `RELOAD`: It can be used for reloading the PgBouncer after a configuration change.
+- `SHUTDOWN`: It can be used for exit the PgBouncer process. Use systemd instead if possible.
+
+## Making an online restart
+
+- You can do an online restart without terminating the connections. PgBouncer launches a new process and loads open sockets from running PgBouncer. After that, old process is stopped and the new process resumes. This way, connections will not be interrupted. This is very useful when upgrading PgBouncer.
+   ```
+   sudo -u postgres pgbouncer -R /etc/pgbouncer/pgbouncer.ini -d
+   ```
+
+## Recommended configurations
+
+- The pooling method can be configured both for the database and globally. (Default: session)
+   ```
+   pool_mode = <session, transaction or statement>
+   ```
+
+- This configuration shows the maximum number of client connections allowed to PgBouncer. You may want to set it large because it can block your connection request. The best practice is to limit this with `pool_size`. Be aware that if you are planning to reach this limit, you may want to increase the number of file descriptors. (Default: 100)
+   ```
+   max_client_conn = max_client_conn + (max pool_size * total databases * total users)
+   ```
+
+- You may want to increase this number to 5 or 10. When the specified `pool_size` is not enough, it uses the reserved pool and logs it. It can be used to determine `pool_size`. (Default: 0)
+   ```
+   reserve_pool_size = 10
+   ```
+
+- Connecting to PostgreSQL through PgBouncer will mask the PostgreSQL IP. This setting adds the client host address and port to the application name. It is very helpful when troubleshooting. (Default: 0)
+   ```
+   application_name_add_host = 1
+   ```
+
+- By default, PgBouncer reuses server connections in LIFO (last-in, first-out). If you are using a TCP load balancer with round-robin behind the PostgreSQL IP address, you may want to enable `server_round_robin` for achieve higher performance. (Default: 1)
+   ```
+   server_round_robin = 1
+   ```
+
+## Recommended authentication method
+By default, PgBouncer comes with `trust` authentication method. The `trust` method can used for testing environment but is not recommended for production. It will accept all connection requests without authentication. For production, `hba` authentication is recommended. With this, you can specify IP addresses with different connection methods:
```

Comment:
```suggestion
By default, PgBouncer comes with `trust` authentication method. The `trust` method can used for testing environment sbut is not recommended for production. It will accept all connection requests without authentication. For production, `hba` authentication is recommended. With this, you can specify IP addresses with different connection methods:
```

Reply:
There is a little typo I guess.

Reply:
Thank you for your time to fix my mistakes. :)